resources/js/features/games/components/+show/GameShowMainRoot.tsx

```diff
@@ -11,14 +12,16 @@ import { GameCommentList } from '../GameCommentList';
 import { GameHeaderSlotContent } from '../GameHeaderSlotContent';
 
 export const GameShowMainRoot: FC = () => {
-  const { game } = usePageProps<App.Platform.Data.GameShowPageProps>();
+  const { game, hasMatureContent } = usePageProps<App.Platform.Data.GameShowPageProps>();
 
   if (!game.badgeUrl || !game.system?.iconUrl) {
     return null;
   }
 
   return (
     <div data-testid="game-show" className="flex flex-col gap-3">
+      {hasMatureContent ? <MatureContentWarningDialog /> : null}
```

Comment:
Can we remember this in a cookie or with a URL parameter? If you hit "yes", but don't select "and don't ask again", any navigation away from the page causes the popup to be displayed again when returning to the page.

i.e:
* Open game page
* Press "Yes, I'm an Adult"
* Go to Supported Game Files or open any achievement
* Hit Back
  [Dialog appears again]

Reply:
I like the URL parameter. In latest, we now add a "mature_content_accepted" param so the state is temporarily remembered in browser history.